### PR TITLE
Adding Shorten Station Name option for configuration

### PR DESCRIPTION
**Pull request by dwardu89**
This is related to Issue #8, I might need some guidance in this.

### CODE REVIEW

File: custom_components/london_tfl/sensor.py [hunk 1]
```diff
@@ -206,3 +210,9 @@ def extra_state_attributes(self):
         attributes['data'] = data
 
         return attributes
+
+    def shortenPlatformName(destinationName, shorten):
+        if CONF_SHORTEN_STATION_NAMES:
+            return destinationName.replace("Underground Station", "").trim()
```
Comment (morosanmihail) [inline custom_components/london_tfl/sensor.py:216]: Could generalise by having a list (in const.py) of strings to remove, and iterating over them here.

File: custom_components/london_tfl/sensor.py [hunk 2]
```diff
@@ -206,3 +210,9 @@ def extra_state_attributes(self):
         attributes['data'] = data
 
         return attributes
+
+    def shortenPlatformName(destinationName, shorten):
+        if CONF_SHORTEN_STATION_NAMES:
```
Comment (morosanmihail) [inline custom_components/london_tfl/sensor.py:215]: This is wrong. It should be `if self.shortenStationNames`. 
`CONF_SHORTEN_STATION_NAMES` always evaluates to true.

File: custom_components/london_tfl/sensor.py [hunk 3]
```diff
@@ -86,14 +88,15 @@ async def async_setup_platform(
                     stop['station'],
                     stop['platform'],
                     stop['max'],
+                    bool(top[CONF_SHORTEN_STATION_NAMES]),
```
Comment (morosanmihail) [inline custom_components/london_tfl/sensor.py:91]: `stop[...` instead of `top[...`

File: custom_components/london_tfl/sensor.py [hunk 4]
```diff
@@ -61,6 +62,7 @@ async def async_setup_entry(
                     stop['station'],
                     stop['platform'] if 'platform' in stop else '',
                     stop['max'] if 'max' in stop else DEFAULT_MAX,
+                    bool(top[CONF_SHORTEN_STATION_NAMES]),
```
Comment (morosanmihail) [inline custom_components/london_tfl/sensor.py:65]: `stop[...` instead of `top[...`

**Review by morosanmihail — CHANGES_REQUESTED** [review]

**Comment by morosanmihail** [comment]
It looks good as a start! A few tiny changes and it'll solve the problem.

**Comment by dwardu89** [comment]
@morosanmihail  I've made those changes, have a look and see what you think.

**Comment by dwardu89** [comment]
I thought of putting "Station" in however there are some names, such as "Battersea Power Station" which would end up being "Battersea Power" potentially an improvement could be that we have a dictionary list of stations and just do a map replace if the shorten list is there. However that might need some manipulations of the API to get the whole list of how it outputs it.

**Review by morosanmihail — APPROVED** [review]

**Pull request merged**